src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int sizeOfArray = 0;
+    private T[] values;
+
+    public ArrayList() {
+        this.values = (T[]) new Object[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (sizeOfArray == values.length) {
+            T[] oldArray = (T[]) new Object[sizeOfArray];
+            System.arraycopy(values, 0, oldArray, 0, values.length);
+            values = (T[]) new Object[grow(sizeOfArray)];
+            System.arraycopy(oldArray, 0, values, 0, oldArray.length);
+        }
+        values[sizeOfArray] = value;
+        sizeOfArray++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > sizeOfArray || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index
+                    + " is greater than array size: " + sizeOfArray);
```

Comment:
```suggestion
            throw new ArrayListIndexOutOfBoundsException("Index: " + index
                    + " is invalid for array size: " + sizeOfArray);
```
index can be negative as well - right?